mlflow/utils/environment.py

```diff
@@ -259,9 +259,8 @@ def _get_pip_version():
              ``__version__`` attribute.
     """
     try:
-        import pip
-
-        return pip.__version__
+        import pkg_resources  # part of setuptools
+        return pkg_resources.get_distribution("pip").version
```

Comment:
We should not use `pkg_resources` because it's deprecated. Can we use `importlib.metadata`?

Reply:
Done. There was another version check that caused issues for me, but importlib seems to do the job there too.